lib/livebook_web/helpers.ex

```diff
@@ -1,44 +1,63 @@
 defmodule LivebookWeb.Helpers do
   use Phoenix.Component
 
+  alias Phoenix.LiveView.JS
+
   alias LivebookWeb.Router.Helpers, as: Routes
 
   alias Livebook.FileSystem
 
   @doc """
-  Renders a component inside the `Livebook.ModalComponent` component.
+  Wraps the given content in a modal dialog.
 
-  The rendered modal receives a `:return_to` option to properly update
-  the URL when the modal is closed.
-  """
-  def live_modal(component, opts) do
-    {modal_opts, opts} = build_modal_opts(opts)
-    modal_opts = [{:render_spec, {:component, component, opts}} | modal_opts]
-    live_component(LivebookWeb.ModalComponent, modal_opts)
-  end
+  When closed, the modal redirects to the given `:return_to` URL.
 
-  @doc """
-  Renders a live view inside the `Livebook.ModalComponent` component.
+  ## Example
 
-  See `live_modal/2` for more details.
+      <.live_modal return_to={...}>
+        <.live_component module={MyComponent}  />
+      </.live_modal>
   """
-  def live_modal(socket, live_view, opts) do
-    {modal_opts, opts} = build_modal_opts(opts)
-    modal_opts = [{:render_spec, {:live_view, socket, live_view, opts}} | modal_opts]
-    live_component(LivebookWeb.ModalComponent, modal_opts)
-  end
-
-  defp build_modal_opts(opts) do
-    path = Keyword.fetch!(opts, :return_to)
-    {modal_class, opts} = Keyword.pop(opts, :modal_class)
+  def modal(assigns) do
+    assigns =
+      assigns
+      |> assign_new(:class, fn -> "" end)
 
-    modal_opts = [
-      id: "modal",
-      return_to: path,
-      modal_class: modal_class
-    ]
+    ~H"""
+    <!-- TODO add phx-remove={JS.transition("fade-out")}, however this
+         doesn't seem to work with child LV rendered inside after opening
+         the modal for the second time -->
+    <div class="fixed z-[10000] inset-0 fade-in">
```

Comment:
@josevalim so we have this piece where we render a child LV inside the modal:

```eex
    <%= if @live_action == :file_settings do %>
      <.modal class="w-full max-w-4xl" return_to={Routes.session_path(@socket, :page, @session.id)}>
        <%= live_render @socket, LivebookWeb.SessionLive.PersistenceLive,
              id: "persistence",
              session: %{
                "session" => @session,
                "file" => @data_view.file,
                "persist_outputs" => @data_view.persist_outputs,
                "autosave_interval_s" => @data_view.autosave_interval_s
              } %>
      </.modal>
    <% end %>
```

If we add here `phx-remove={JS.transition("fade-out")}`, then opening and closing the modal once works, but opening it again no longer shows the content. Any ideas?

Reply:
Can you walk me thru how to cause the issue? With your branch, if I add this to the root modal:

        <div class="fixed z-[10000] inset-0 fade-in" phx-remove={JS.transition("fade-out")}> 

then visit http://localhost:4000/settings/add-file-system and click away, browser back button, or click close, it works over and over. Is there another page that I should be trying?

Reply:
Open any notebook, and in the right-bottom corner click the save icon, then close the dialog, then click the icon again.

It needs to be this particular modal, because its content is LV child and not a component as in other cases.

Reply:
This was a bug on the LV side. I just pushed a fix and a 0.17.5 release. We weren't finding nested LV children when deferring removes with phx-remove and calling destroy on them, so that modal child remained alive, thus it didn't re-mount because the client thought it already was on the page. Will cut a new release

Reply:
Beautiful, thank you!! :cat: